### PR TITLE
WIP: CLI command for publishing and creating a new bootstrap

### PR DESCRIPTION
**Pull request by jdeepee**
WIP because the code needs to be cleaned up, and the logs improved. Otherwise it is functional and has been used to succesfully generate a new bootstrap seed.

Note: this spawns its own ad4m-host instance, and moves agent data around. This is added for convience since when updating the language language, if you did not automate this process you would have to manually create the temporary publishing bootstrap seed, copy agent and then run ad4m-host init + serve.  With this structure that is all handled via the cli.

Would like to hear you first thoughts @lucksus

### CODE REVIEW

File: cli/src/languages.rs
Comment (lucksus) [inline cli/src/languages.rs]: I don't think this needs to be the case, no? We don't need to use `executor_data_path()` below, but could just use a temp directory.

File: cli/seed_proto.json
Comment (lucksus) [inline cli/seed_proto.json]: `perspectiveDiffSync` is different to the other languages in that it is the name of a specific link language implementation, while the other languages need to have exactly one - and `agentLanguage` for instance describes the role of that language.

We could have this be an array with the name "linkLanguages" and then we could really fill the `knownLinkLanguages` array with multiple.

**Review by lucksus — COMMENTED** [review]
This is awesome, and I'd be happy to merge after some cleanup. I have two small suggestions though, that I wouldn't consider blocking the merge..

File: cli/src/languages.rs
Comment (jdeepee) [reply cli/src/languages.rs]: Ah yes I think that is possible and would avoid having to backup and delete existing agents.

File: cli/seed_proto.json
Comment (jdeepee) [reply cli/seed_proto.json]: yeah right; ill rename the key to linkLanguages and make it an array

**Review by lucksus — APPROVED** [review]
perfect

**Pull request merged**
